packages/flutter_driver/lib/src/retry.dart

```diff
@@ -28,20 +32,26 @@ Future<dynamic> retry(Action action, Duration timeout,
   while(!success && sw.elapsed < timeout) {
     try {
       result = await action();
-      success = true;
+      if (predicate == null || predicate(result) == true) {
```

Comment:
`predicate` is supplied by 3rd-party code and could result in `null`. Are we ok with throwing in that case?

Braces done.